### DISCUSSION TITLE
passing proprietary capabilities

### DISCUSSION THREAD

**Discussion by tallevy22** [seleniumbase/SeleniumBase#1071] [Q&A] [answered]
im trying to use seleniumbase against Perfectomobile cloud, and need to.
desired_cap = {
    'platformName': 'Windows',
    'platformVersion': '10',
    'browserName': 'Chrome',
    'browserVersion': '87',
    'location': 'US East',
    'resolution': '1024x768',
    'securityToken': 'XXXXX' }

**Comment by mdmintz** [comment, marked as answer]
You can pass any capabilities that you'd like to as long as they are supported by the Selenium Grid that you're connecting to.
https://github.com/seleniumbase/SeleniumBase/blob/master/examples/capabilities/ReadMe.md

**Reply by tallevy22** [reply]
when i checked the address that was rendered I get 'https://demo.perfectomobile.com/nexperience/perfectomobile:4444/wd/hub'
it seems to insert the port in the wrong place

**Reply by mdmintz** [reply]
@tallevy22 try this:
``--server=demo.perfectomobile.com/nexperience/perfectomobile --port=443 --protocol=https``
(Change the port as needed. Likely 443 due to https.)

**Reply by tallevy22** [reply]
what i was referring to is that the URL rendered by the code inserts the port number in the wrong place, usually the port is supposed to be added : https://<cloud>:<port>/xxxx/wd/hub
but instead the port is added 'https://demo.perfectomobile.com/nexperience/perfectomobile:4444/wd/hub'
I'm not refferinng inn my question to the number port.
hope I the question is clear :-)

**Reply by mdmintz** [reply]
I understand the issue now. I'll ship a fix in the next SeleniumBase release (it'll be today).

**Reply by tallevy22** [reply]
wow, thanks!!!

**Reply by mdmintz** [reply]
@tallevy22 I just shipped two new versions of seleniumbase, and one of them has the fix, so just download the latest, give it a try, and let me know if you still have any issues with it.

**Reply by tallevy22** [reply]
its rendering correctly now, thank you
